src/crimson/os/seastore/object_data_handler.h

```diff
@@ -35,30 +51,30 @@ struct ObjectDataBlock : crimson::os::seastore::LogicalCachedExtent {
     return TYPE;
   }
 
-  ceph::bufferlist get_delta() final {
-    /* Currently, we always allocate fresh ObjectDataBlock's rather than
-     * mutating existing ones. */
-    ceph_assert(0 == "Should be impossible");
+  void add_contents(bufferlist bl, uint64_t offset) {
+    delta.push_back({offset, bl.length(), bl});
   }
 
-  void apply_delta(const ceph::bufferlist &bl) final {
-    // See get_delta()
-    ceph_assert(0 == "Should be impossible");
-  }
+  ceph::bufferlist get_delta() final;
+
+  void apply_delta(const ceph::bufferlist &bl) final;
 };
 using ObjectDataBlockRef = TCachedExtentRef<ObjectDataBlock>;
 
 class ObjectDataHandler {
 public:
   using base_iertr = TransactionManager::base_iertr;
 
-  ObjectDataHandler(uint32_t mos) : max_object_size(mos) {}
+  ObjectDataHandler(uint32_t mos) : max_object_size(mos),
+    partial_overwrite(
+      crimson::common::get_conf<Option::size_t>("seastore_partial_overwrite")) {}
 
   struct context_t {
     TransactionManager &tm;
     Transaction &t;
     Onode &onode;
     Onode *d_onode = nullptr; // The desination node in case of clone
+    size_t po_size = 0;
```

Comment:
The reason why I added the po_size to the context_t is to hand over po_size as an argument of the `prepare_ops_list()`. 
At first, I used the ObjectDataHandler.partial_overwrite for that instead of the po_size, but I encountered a freed memory access because ObjectDataHandler is freed after issuing requests in the case of `object_data_handler_test_t.write()`.

We probably can alternatively use the partial-overwrite size in crimson.yaml.in by using crimson::common::get_conf(). But, this way is a bit expensive because it might load conf every time the write is called.

Reply:
> At first, I used the ObjectDataHandler.partial_overwrite for that instead of the po_size, but I encountered a freed memory access because ObjectDataHandler is freed after issuing requests in the case of object_data_handler_test_t.write().

Looks `ctx.po_size` is used by `prepare_ops_list()` which is part of `ObjectDataHandler::trim_data_reservation()` or `ObjectDataHandler::overwrite()` procedure.

So it doesn't look reasonable for the test to free `ObjectDataHandler` before it's operation is completed, right?

Reply:
Right. How about replacing the existing code  in test_object_data_handler.cc with the following?

As is:
```
      return ObjectDataHandler(MAX_OBJECT_SIZE).write(
        ObjectDataHandler::context_t{
          *tm,
          t,
          *onode,
        },
        offset,
        bl);
```

To be:

```
      return seastar::do_with(
        std::move(bl),
        ObjectDataHandler(MAX_OBJECT_SIZE),
        [=, this, &t](auto &bl, auto &objhandler) {
          return objhandler.write(
            ObjectDataHandler::context_t{
              *tm,
              t,
              *onode,
            },
            offset,
            bl);
        });
```

Reply:
Looks reasonable.